app/views/teachers/TeacherStudentView.coffee

```diff
@@ -95,11 +79,122 @@ module.exports = class TeacherStudentView extends RootView
       $("[id|='visualisation']").hide()
       $(selected).show()
 
+  calculateStandardDev: ->
+    return unless @courses.loaded and @levels.loaded and @sessions?.loaded and @levelData
+
+    @courseComparisonMap = []
+    for versionedCourse in @classroom.get('courses') ? []
+      course = _.find @courses.models, (c) => c.id is versionedCourse._id
+      numbers = []
+      studentRate = 0
+      members = 0 #this is the COUNT for our standard deviation
+      for member in @classroom.get('members')
+        NUMBER = 0
+        memberBit = 0
+        for versionedLevel in versionedCourse.levels
+          for session in @sessions.models
+            if session.get('level').original == versionedLevel.original and session.get('creator') == member
+              # TODO IMPORTANT: only add number if @studentID in levelProgressMap has complete or started for the corresponding level
```

Comment:
This can be done with the above comment's code slightly changed:

```
for session in @sessions.where({ creator: member }).filter (s) ->
    s.get('level').original is versionedLevel.original and s.get('state')?.complete
```